ansible_inventory_server/maasrest.py

```diff
@@ -21,13 +21,34 @@
                                             filter_ip_addresses)
 
 
+def maas_filter_ip_addresses(data, kwargs):
+    """Filters IP addresses using network interface name and/or subnet"""
+
+    ip_addresses = []
+
+    try:
+        interface_filter = kwargs.get('interface')
+        for interface in data['interface_set']:
+            if interface_filter and interface['name'] != interface_filter:
+                continue
+
+            links = interface.get('links') or []
+            links += interface.get('discovered') or []
+            ip_addresses.extend([link['ip_address'] for link in links])
+
+    except (KeyError, NameError):
+        pass
+
+    return filter_ip_addresses(ip_addresses, kwargs)
+
+
 def filter_maas_machine_info(machine, kwargs):
     """Keeps only useful machine information"""
     return {
         'system_id': machine['system_id'],
         'fqdn': machine['fqdn'],
         'hostname': machine['hostname'],
-        'ip_addresses': filter_ip_addresses(machine['ip_addresses'], kwargs),
+        'ip_addresses': maas_filter_ip_addresses(machine, kwargs),
```

Comment:
machine['interface_set'] would be enough because this is the only part of machine you are using